browser_use/agent/playwright_script_generator.py

```diff
@@ -0,0 +1,734 @@
+        print(f"  Warning: Action '{action_type}' failed with original selector ({repr(selector)}): {e}. Starting fallback...")
+
+        # Fallback only works for XPath selectors
+        if not selector.startswith('xpath='):
+            # Raise error immediately if not XPath, as fallback won't work
+            raise PlaywrightActionError(f"Action '{action_type}' failed. Fallback not possible for non-XPath selector: {repr(selector)}. ({step_info})")
+
+        xpath_parts = selector.split('=', 1)
+        if len(xpath_parts) < 2:
+             raise PlaywrightActionError(f"Action '{action_type}' failed. Could not extract XPath string from selector: {repr(selector)}. ({step_info})")
+        xpath = xpath_parts[1] # Correctly get the XPath string
+
+        segments = [seg for seg in xpath.split('/') if seg]
+
+        for i in range(1, min(MAX_FALLBACKS + 1, len(segments))):
+            trimmed_xpath_raw = '/'.join(segments[i:])
+            fallback_xpath = f'xpath=//{trimmed_xpath_raw}'
+
+            print(f'    Fallback attempt {i}/{MAX_FALLBACKS}: Trying selector: {repr(fallback_xpath)}')
+            try:
+                locator = page.locator(fallback_xpath).first
+                if action_type == 'click':
+                     await locator.click(timeout=FALLBACK_TIMEOUT)
+                elif action_type == 'fill' and text is not None:
+                    try:
+                        await locator.clear(timeout=FALLBACK_TIMEOUT)
+                        await page.wait_for_timeout(100)
+                    except Exception as clear_error:
+                         print(f"    Warning: Failed to clear field during fallback ({step_info}): {clear_error}")
+                    await locator.fill(text, timeout=FALLBACK_TIMEOUT)
+
+                print(f"    Action '{action_type}' successful with fallback selector: {repr(fallback_xpath)}")
+                await page.wait_for_timeout(500)
+                return # Successful exit after fallback
+            except Exception as fallback_e:
+                print(f'    Fallback attempt {i} failed: {fallback_e}')
+                if i == MAX_FALLBACKS:
+                    # Raise exception after exhausting fallbacks
+                    raise PlaywrightActionError(f"Action '{action_type}' failed after {MAX_FALLBACKS} fallback attempts. Original selector: {repr(original_selector)}. ({step_info})")
+
+    # This part should not be reachable if logic is correct, but added as safeguard
+    raise PlaywrightActionError(f"Action '{action_type}' failed unexpectedly for {repr(original_selector)}. ({step_info})")
+
+"""
+        return [
+            "import asyncio",
+            "import json",
+            "import os",
+            "import sys",
+            "from pathlib import Path", # Added Path import
+            "import urllib.parse", # Needed for search_google
+            "from playwright.async_api import async_playwright, Page, BrowserContext", # Added BrowserContext
```

Comment:
FYI we use patchright now so this should also be updated
```suggestion
            "from patchright.async_api import async_playwright, Page, BrowserContext", # Added BrowserContext
```